### PR TITLE
ci: run examples and integration tests in parallel

### PR DESCRIPTION
**Pull request by gakonst**
As title

### CODE REVIEW

File: ops/scripts/load-images.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Script for loading all the docker images in CI
```
Comment (tynes) [inline ops/scripts/load-images.sh:1]: This should have a `#!/bin/bash` at the top

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -36,6 +36,32 @@ jobs:
         working-directory: ./ops
         run: ./scripts/build-ci.sh
 
+      - name: Save the images
+        working-directory: ./ops
+        run: ./scripts/save-images.sh
```
Comment (tynes) [inline .github/workflows/integration.yml:41]: Will this collide with other PRs? ie another PR is submitted that saves images and then retriggering the CI causes different images to be used?

**Comment by tynes** [comment]
We will need to update the required checks to include the new name - it is no longer `integration` but instead `integration_tests`. We should also add one for the `examples`. The `integration` check above will wait forever

File: .github/workflows/integration.yml (hunk 2 above)
Comment (gakonst) [reply .github/workflows/integration.yml:41]: I don't think so, each one is in its own unique runner, no?

**Pull request closed without merging**